### DISCUSSION TITLE
Are the following initializations supposed to be different?

### DISCUSSION THREAD

**Discussion by LinZhihao-723** [nlohmann/json#4096] [Q&A] [answered]
I'm initializing a json object using the following two different alternatives in c++17:
```
nlohmann::json A = nlohmann::json::parse(metadata_buffer.begin(), metadata_buffer.end()); // Object A
nlohmann::json B{nlohmann::json::parse(metadata_buffer.begin(), metadata_buffer.end())}; //Object B
```
They are initialized with the same data; however, A differs from B.
Basically, B is equivalent to `[A]`, where everything parsed is considered the first element in the list. Is this behavior expected?

**Comment by gregmarr** [comment]
Supposed to be, not really, but they are.  Unfortunately, it can't easily be fixed.  In  fact, the behavior varies depending on the compiler, so avoiding B is recommended.

**Reply by LinZhihao-723** [reply]
So suppose I have a vector of bytes, the correct way to convert them into JSON object is using A? Or is there any more proper way such as directly calling a constructor?

**Reply by gregmarr** [reply, marked as answer]
Yes, A is the proper form.  You can even simplify it a bit:

    auto A = nlohmann::json::parse(metadata_buffer.begin(), metadata_buffer.end());
